From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Marcelo Ricardo Leitner <marcelo.leitner@gmail.com>
Date: Fri, 21 Dec 2018 10:25:19 -0200
Subject: [PATCH] withsctp: to not reuse PACKAGE_VERSION as lib version

Currently and likely in the future, they resolv to the same value, but
conceptually speaking they are different things. As we introduced
dedicated variables for libsctp and libwithsctp versioning, lets avoid
possible confusion and expand them instead of reusing the
PACKAGE_VERSION one.

Fixes: #28
Signed-off-by: Marcelo Ricardo Leitner <marcelo.leitner@gmail.com>
---
 Makefile.rules           | 4 +++-
 src/withsctp/withsctp.in | 2 +-
 2 files changed, 4 insertions(+), 2 deletions(-)

diff --git a/Makefile.rules b/Makefile.rules
index d3693fa..9488228 100644
--- a/Makefile.rules
+++ b/Makefile.rules
@@ -16,4 +16,6 @@ edit = @sed \
 	-e "s|\@bindir\@|$(bindir)|" \
 	-e "s|\@libdir\@|$(libdir)|" \
 	-e "s|\@PACKAGE\@|$(PACKAGE)|" \
-	-e "s|\@PACKAGE_VERSION\@|$(PACKAGE_VERSION)|"
+	-e "s|\@LIBWITHSCTP_CURRENT\@|$(LIBWITHSCTP_CURRENT)|" \
+	-e "s|\@LIBWITHSCTP_REVISION\@|$(LIBWITHSCTP_REVISION)|" \
+	-e "s|\@LIBWITHSCTP_AGE\@|$(LIBWITHSCTP_AGE)|"
diff --git a/src/withsctp/withsctp.in b/src/withsctp/withsctp.in
index fda5ebc..ef4f7bb 100644
--- a/src/withsctp/withsctp.in
+++ b/src/withsctp/withsctp.in
@@ -2,7 +2,7 @@
 # -*- sh -*-
 LIBDIR=@libdir@/@PACKAGE@
 BINDIR=@bindir@
-LIBVER=@PACKAGE_VERSION@
+LIBVER=@LIBWITHSCTP_CURRENT@.@LIBWITHSCTP_AGE@.@LIBWITHSCTP_REVISION@
 export LD_PRELOAD=${LIBDIR}/libwithsctp.so.${LIBVER}
 if ! ${BINDIR}/checksctp 2> /dev/null
 then
